libraries/splunk_password.rb

```diff
@@ -0,0 +1,52 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: splunk_password.rb
+
+require 'base64'
+
+# Module contains different functions to encrypt and decrypt splunk passwords
+module CernerSplunk
+  # Encrypts the password before writing into config files. As of now all the passwords
+  # needs to be XORed except for the sslKeyFilePassword. The boolean
+  # parameter xor controls the XOR logic.
+  def self.splunk_encrypt_password(plain_text, splunk_secret, xor = true)
+    fail 'string to be encrypted cannot be empty/nil' if plain_text.nil? || plain_text.empty?
```

Comment:
I didn't see a reason why someone should be given the option to encrpyt/decrypt an empty string and hence added this check.